### ISSUE TITLE
Drop downs do not always translate

### ISSUE THREAD

**Issue by NAllred91** [mo4islona/node-blockly#19]
You can reproduce this in your live demo, http://mo4islona.github.io/blockly/

First, load the page and change the language to french.  Open the the Loops flyout to see that the while block has been translated to french.  Then close the flyout and translate back to english.  Check the Loops flyout again and still see french in the While block.

It seems like these blocks are getting stuck on whatever language they are set to when you first look at them.

**Comment by NAllred91** [comment]
I've got a hacky way I think I can fix this in my own fork of node-blockly.  I'm not sure how what the correct way to fix this is yet.

**Comment by mo4islona** [comment]
Yeah, Blockly caches some blocks then loads it at first time. I m not sure that it possible fix that normally. Some dirty hack is needed(

**Comment by mo4islona** [comment]
Check please 1.0.29

**Comment by mo4islona** [comment]
I think the problem is inside Blockly. Some blocks translates back correctly some not

**Comment by NAllred91** [comment]
Ah yea... At first I thought it was only drop downs but now I see its all pretty random.  I'll take a look at blockly at some point and see if I can do a PR there.

For now, I may just need to fork node-blockly and make `require('node-blockly/browser') not export a self invoking function anymore.  I think if I make it just export a function I can do

```
getBlockly = require('node-blockly/browser')
Blockly = getBlockly()
```

and call `getBlockly` whenever I need a new instance of blockly, or something like that..

**Comment by NAllred91** [comment]
Turns out my work around wasn't quite enough...

This line causes problems..
https://github.com/mo4islona/node-blockly/blob/master/browser.js#L12

Since `setLocale` is always called initially with english, so even when you clear the require cache it still gets called with english from the start and that messes up some blocks.

I'm not sure if my current solution is good enough for a PR, but I changed browser.js to this
```
var getBlockly = function(languageCode) {
  delete require.cache[require.resolve('node-blockly/lib/blockly_compressed_browser')]
  delete require.cache[require.resolve('node-blockly/lib/blocks_compressed_browser')]
  var Blockly = require('./lib/blockly_compressed_browser');

  Blockly._setLocale = function(locale) {
    Blockly.Msg = Object.assign(locale, Blockly.Msg);
    Blockly.Msg = Blockly.Msg();
  }

  Blockly.utils.getMessageArray_ = function () {
    return Blockly.Msg
  }

  Blockly._setLocale(require('./lib/i18n/' + languageCode))

  Blockly.Blocks = Object.assign(Blockly.Blocks, require('./lib/blocks_compressed_browser')(Blockly));

  Blockly.JavaScript = require('./lib/javascript_compressed')(Blockly);
  Blockly.Lua = require('./lib/lua_compressed')(Blockly);
  Blockly.Dart = require('./lib/dart_compressed')(Blockly);
  Blockly.PHP = require('./lib/php_compressed')(Blockly);
  Blockly.Python = require('./lib/python_compressed')(Blockly);

  return Blockly
}

class Blockly {
  constructor() {
      Object.assign(this, getBlockly('en'))
  }

  setLanguage = (languageCode) => {
      Object.assign(this, getBlockly(languageCode))
  }
}

module.exports =  new Blockly()
```

Then, to change the language I do this
```
import Blockly from 'node-blockly/browser'

var changeLanguage = (languageCode) => {
    Blockly.setLanguage(languageCode)
    var currentXML = this._getWorkspaceXML()
    this._workspace.dispose()
    this._workspace = Blockly.inject(config.blocklyElementId, {
                toolbox: getToolbox(this._appManager),
                grid: config.blocklyGridOptions,
            })
    this._setWorkspaceXML(currentXML)
}
```


With this solution, `Blockly._setLocale` only ever gets called on a given instance of `Blockly` once.

**Comment by NAllred91** [comment]
But this solution is nice because I can change the language without doing the weird cache invalidation in my own code.

If I made a PR using this pattern and cleaned it up some keeping `Blockly.setLocale` as the public method, I don't think it would break existing code that uses `setLocale`?  Let me know if you're interested in a PR with this pattern.
